router/worker.go

```diff
@@ -897,6 +897,17 @@ func (w *worker) retryLimitReached(status *jobsdb.JobStatusT) bool {
 		}
 	}
 	respStatusCode, _ := strconv.Atoi(status.ErrorCode)
+
+	if status.JobParameters != nil {
+		if jobRunID := gjson.GetBytes(status.JobParameters, "source_job_run_id").Str; jobRunID != "" {
+			return respStatusCode >= 500 &&
+				respStatusCode != types.RouterTimedOutStatusCode &&
+				respStatusCode != types.RouterUnMarshalErrorCode && // 5xx errors
+				time.Since(firstAttemptedAtTime) > w.rt.reloadableConfig.sourcesRetryTimeWindow.Load() &&
+				status.AttemptNum >= w.rt.reloadableConfig.maxFailedCountForSourcesJob.Load() // retry time window exceeded
+		}
+	}
+
```

Comment:
instead of repeating the same logic twice, we can resolve the thresholds first and perform the logic next, once
```suggestion
	maxFailedCountForJob := w.rt.reloadableConfig.maxFailedCountForJob.Load()
	retryTimeWindow := w.rt.reloadableConfig.retryTimeWindow.Load()
	if gjson.GetBytes(status.JobParameters, "source_job_run_id").Str != "" {
		maxFailedCountForJob = w.rt.reloadableConfig.maxFailedCountForSourcesJob.Load()
		retryTimeWindow = w.rt.reloadableConfig.sourcesRetryTimeWindow.Load()
	}

	return (respStatusCode >= 500 &&
		respStatusCode != types.RouterTimedOutStatusCode && respStatusCode != types.RouterUnMarshalErrorCode) && // 5xx errors
		time.Since(firstAttemptedAtTime) > retryTimeWindow &&
		status.AttemptNum >= maxFailedCountForJob // retry time window exceeded

```